### PR TITLE
Add Chunk Filenaming Support in Webpack Config

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
- create reusable webpack entries
- enable chunk file naming support

Both are for code splitting from an application level.

### CODE REVIEW

**Review by bjankord — APPROVED** [review]

File: config/webpack/webpack.config.js [hunk 1]
```diff
@@ -9,30 +9,25 @@ const TerserPlugin = require('terser-webpack-plugin');
...
-    themeFile,
     staticOptions,
   } = options;
 
   const production = argv.p;
-  let filename = production ? '[name]-[chunkhash]' : '[name]';
-  filename = argv['output-filename'] || filename;
+  const fileNameStategy = production ? '[name]-[chunkhash]' : '[name]';
+  const chunkFilename = argv['output-chunk-filename'] || fileNameStategy;
+  const filename = argv['output-filename'] || fileNameStategy;
```
Comment (mjhenkes) [inline config/webpack/webpack.config.js:25]: Why do we need to customize this? I'm concerned with how complicated this webpack config has grown.

**Comment by emilyrohrbough** [comment]
Closing this PR. Talked with @mjhenkes and this added complexity isn't needed.

**Comment by emilyrohrbough** [comment]
Talked with @mjhenkes again. Should have mentioned these were webpack argv options that were exposes/available via CLI as well

**Review by mjhenkes — APPROVED** [review]

**Comment by emilyrohrbough** [comment]
Validation PR: https://github.com/cerner/terra-dev-site/pull/186

**Review by ryanthemanuel — APPROVED** [review]

**Review by StephenEsser — APPROVED** [review]

**Pull request merged**